lib/actions.js

```diff
@@ -1616,16 +1633,34 @@ exports.waitFor = function() {
 /**
  * Waits for a function to evaluate to a given value on the given page
  * @param {Page | undefined} page
- * @param {function} fn
+ * @param {function | object} optsOrFn - If optsOrFn is a function, use the classic signature waitForPage(page, fn, arg1, arg2, value), If arg is an object, use waitForPage(page, options)
+ * @param {array} optsOrFn.args - arguments of fn
+ * @param {function} optsOrFn.fn - function to evaluate
+ * @param {function} optsOrFn.fname - function name (can be set with fn.name on the classic signature)
+ * @param {*} optsOrFn.value - expected value of function
+ * @param {number} [optsOrFn.timeout=null] - timeout in ms
  * @param {...*} [arguments]
  * @param {*} value
  * @emits Horseman#TimeoutError
  */
-function waitForPage(page, fn) {
+function waitForPage(page, optsOrFn) {
 	var self = this;
-	var args = Array.prototype.slice.call(arguments);
-	var value = args.pop();
-	var fname = fn.name || '<anonymous>';
+	var args, value, fname, timeout = self.options.timeout, fn;
+
+	if(typeof optsOrFn === "function"){
```

Comment:
The indentation in this `if` and `else` is messed up. Sorry but I am neurotic about this sort of thing.